File: Chapter17/17-16 ObjC/ObjCLazyLoadingProperty/ObjCLazyLoadingProperty/Person.m
//
//  Person.m
//  ObjCLazyLoadingProperty
//
//  Created by Keun young Kim on 2014. 10. 24..
//  Copyright (c) 2014년 Keun young Kim. All rights reserved.
//

#import "Person.h"

@implementation Person

- (Contact*)contact
{
    if (_contact == nil) {
        _contact = [Contact new];
    }
    
    return _contact;
}

@end

File: Chapter21/21-20/SwiftPrintableProtocol/SwiftPrintableProtocol/Person.swift
//
//  Person.swift
//  SwiftPrintableProtocol
//
//  Created by Keun young Kim on 2014. 10. 24..
//  Copyright (c) 2014년 Keun young Kim. All rights reserved.
//

import Foundation



class Person: Printable {
    var name = "James"
    var age = 100
    
    var description: String {
        return name + "(\(age))"
    }
}